### DISCUSSION TITLE
Cannot modify HTML body

### DISCUSSION THREAD

**Discussion by creamtart** [mitmproxy/mitmproxy#6258] [General]
Hello people. I am having issues modifying the HTML response content body when using MITM in reverse mode. I am following the guide exactly as it says and cannot get it to work. 

I am is to replace the content of an HTML tag. From ``<h1>OLD</h1> to <h2>NEW</h2>`` 

I only managed to get it working using the code bellow but it allows me to modify only one object at a time: 
```python
def modify(data: bytes) -> Union[bytes, Iterable[bytes]]:
         return data.replace(b"``<h1>OLD</h1>``", b"``<h2>NEW</h2>``")

  def responseheaders(flow):
        flow.response.stream = modify
```
When trying to use the proper method using the code bellow I get no results: 
```python
from mitmproxy import http


def response(flow: http.HTTPFlow) -> None:
      if flow.response and flow.response.content:
          flow.response.content = flow.response.content.replace(
              b"<h1>OLD</h1>", b"<h2>NEW</h2>"
           )
```

Would appreciate any suggestions.

**Comment by mhils** [comment]
Two things: First, the [modify_body](https://docs.mitmproxy.org/stable/overview-features/#modify-body) option may just be what you are looking for. Second, is there a reason you're using 
```
b"``<h1>OLD</h1>``"
```
and not
```
b"<h1>OLD</h1>"
```
? :)

**Reply by creamtart** [reply]
Hey. Thank you for replying. That's just a github thing, I messed up the code box, my code does not have the "``" characters. I just can't get it to work for some reason. Really weird situation.

**Reply by creamtart** [reply]
Update: I believe I had a corrupt install as I set up a new virtual environment, reinstalled MITM, used the exact same script and everything functioned perfectly. Thank you.

**Reply by rosydawn6** [reply]
I found this and a few other links looking for information on this topic.  Several attempts were not producing the expected results.  I was using stream_large_bodies=1 and after rereading the docs more https://docs.mitmproxy.org/stable/concepts-options/#stream_large_bodies found that when using stream_large_bodies=1 the body is not stored and hence cannot be modified.
